### PR TITLE
Enhancing Makefile features

### PR DESCRIPTION
**Pull request by msoares94**
Adding new behaviors and adjusting stop behavior.
    
Readjusted build and configure behavior as it makes more sense to copy the .env.example only if the project doesn't already have an .env

Updating the Readme.md to reflect the changes, it makes sense to run configure before the build

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -7,13 +7,23 @@ help:
 start: ## Spin up the containers and run the app UI in watch mode
 	docker compose up -d
 
-stop: ## Shut down the containers
+stop: ## Stop the containers
```
Comment (WendellAdriel) [inline Makefile:10]: Why would you need to stop and not remove the containers in a dev env?

File: Makefile [hunk 2]
```diff
@@ -7,13 +7,23 @@ help:
...
+	&& rm -f docker/redis/data/dump.rdb
+
+status: ## Status the containers
+	docker compose ps
+
+down: ## Shut down the containers
 	docker compose down \
 	&& rm -f docker/redis/data/dump.rdb
 
+down-with-vol: ## Shut down the containers and removed volumes
```
Comment (WendellAdriel) [inline Makefile]: No need for a new command, you can add ARGS to the other command like we do for other commands so the args can be used

File: Makefile [hunk 3]
```diff
@@ -23,7 +33,8 @@ db-stop: ## Shut down the DB container
 
 ##@ [Application]
 configure: ## Configures the application when setting it for the first time
-	make install \
+	[ -f .env ] || cp .env.example .env \
```
Comment (WendellAdriel) [inline Makefile]: This should be in the build command instead because it doesn't make sense to run the configure before the build since the make install and make art commands need the images built to run

File: README.md
Comment (WendellAdriel) [inline README.md:1]: it doesn't make sense to run the configure before the build since the make install and make art commands need the images built to run. Revert the changes here

**Review by WendellAdriel — CHANGES_REQUESTED** [review]
I added some comments for you to check

File: Makefile (hunk 1 above)
Comment (msoares94) [reply Makefile:10]: It makes sense when you have a large database with migrations and seeders already running at some point and you just want to stop your environment to continue working the next day.

File: Makefile (hunk 2 above)
Comment (msoares94) [reply Makefile]: I added it because the command already defined in full makes it easier to use by the IDE, thus giving more proactivity and efficiency.

If you think you shouldn't have it, we can remove it.

File: Makefile (hunk 3 above)
Comment (msoares94) [reply Makefile]: This is where the problem occurs, the build being executed before configure will present an error because the `docker/app/Dockerfile` mentions:

`COPY .env .env`


Soon the error will occur because the file does not exist yet, hence the inversion of the order as well as the adjustment to verify whether or not the file exists before copying the .env.example

File: README.md
Comment (msoares94) [reply README.md:1]: Commented above

File: Makefile (hunk 2 above)
Comment (WendellAdriel) [reply Makefile]: Rename it to down-v only then, please

File: Makefile (hunk 3 above)
Comment (WendellAdriel) [reply Makefile]: But the make configure needs the images built. I don’t understand what’s the problem here since it already copies the file in the build

File: README.md
Comment (WendellAdriel) [reply README.md:1]: Replied above

File: Makefile (hunk 3 above)
Comment (msoares94) [reply Makefile]: Well then, this statement assumes that the .env exists, but if the project was just cloned and if you try to run the build, it will try to build the image and fail at that point.

File: Makefile (hunk 2 above)
Comment (msoares94) [reply Makefile]: It really makes sense to pass on as an argument, considering that for productivity and efficiency the removal of the volume of data does not enter into this requirement, since such an action requires greater attention.

I'll leave it as an argument as suggested

File: Makefile (hunk 3 above)
Comment (WendellAdriel) [reply Makefile]: The make build command first creates the .env and then run the build command with Docker. IDK why this would fail when trying to run the build command first

Comment (msoares94) [reply Makefile]: Okay, really, my mistake.

I'll just add the validation, to copy if it doesn't exist.

**Review by WendellAdriel — APPROVED** [review]
Thanks for the contribution! 💪
